sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs

```diff
@@ -373,14 +373,39 @@ private async Task<IReadOnlyList<ServiceBusReceivedMessage>> ReceiveMessagesAsyn
                 // The session won't be closed in the case that MaxConcurrentCallsPerSession > 1, but with concurrency, it is not possible to guarantee ordering.
                 if (_isSessionReceiver && (!_isProcessor || SessionId != null) && messageList.Count < maxMessages)
                 {
-                    await link.DrainAsyc(cancellationToken).ConfigureAwait(false);
-
-                    // These workarounds are necessary in order to resume prefetching after the link has been drained
-                    // https://github.com/Azure/azure-amqp/issues/252#issuecomment-1942734342
-                    if (_prefetchCount > 0)
+                    try
```

Comment:
I wonder if we should extract this entire try/catch block into a new method called like `DrainLinkAsync` or similar.   With all the logging in there, its chunky enough a separate method would improve readability here.   Thoughts?